PR 33000

jianyuh opened the pull request:
For test of https://github.com/pytorch/pytorch/pull/33000

jianyuh committed:
[pt][fbgemm] Turn on USE_FBGEMM on Windows env

As Title says

Differential Revision: [D19738268](https://our.internmc.facebook.com/intern/diff/D19738268/)

[ghstack-poisoned]

jianyuh committed:
Update on "[pt][fbgemm] Turn on USE_FBGEMM on Windows env"


As Title says

Differential Revision: [D19738268](https://our.internmc.facebook.com/intern/diff/D19738268/)

[ghstack-poisoned]

jianyuh committed:
Update on "[pt][fbgemm] Turn on USE_FBGEMM on Windows env"


As Title says

Differential Revision: [D19738268](https://our.internmc.facebook.com/intern/diff/D19738268/)

[ghstack-poisoned]

jianyuh committed:
Update on "[pt][fbgemm] Turn on USE_FBGEMM on Windows env"


As Title says

Differential Revision: [D19738268](https://our.internmc.facebook.com/intern/diff/D19738268/)

[ghstack-poisoned]

jianyuh committed:
Update on "[pt][fbgemm] Turn on USE_FBGEMM on Windows env"


As Title says. FBGEMM has recently added the support for Windows.

Differential Revision: [D19738268](https://our.internmc.facebook.com/intern/diff/D19738268/)

[ghstack-poisoned]

jianyuh committed:
Update on "[pt][fbgemm] Turn on USE_FBGEMM on Windows env"


As Title says. FBGEMM has recently added the support for Windows.

Differential Revision: [D19738268](https://our.internmc.facebook.com/intern/diff/D19738268/)

[ghstack-poisoned]

jianyuh committed:
Update on "[pt][fbgemm] Turn on USE_FBGEMM on Windows env"


As Title says. FBGEMM has recently added the support for Windows.

Differential Revision: [D19738268](https://our.internmc.facebook.com/intern/diff/D19738268/)

[ghstack-poisoned]

peterjc123 committed:
debug avx512

peterjc123 committed:
add missing flag for avx512

peterjc123 committed:
Remove flags

peterjc123 committed:
Switch to VS 14.12

jianyuh committed:
Update on "[pt][fbgemm] Turn on USE_FBGEMM on Windows env"


As Title says. FBGEMM has recently added the support for Windows.

Differential Revision: [D19738268](https://our.internmc.facebook.com/intern/diff/D19738268/)

[ghstack-poisoned]

jianyuh committed:
Update on "[pt][fbgemm] Turn on USE_FBGEMM on Windows env"


As Title says. FBGEMM has recently added the support for Windows.

Differential Revision: [D19738268](https://our.internmc.facebook.com/intern/diff/D19738268/)

[ghstack-poisoned]

jianyuh committed:
Update on "[pt][fbgemm] Turn on USE_FBGEMM on Windows env"


As Title says. FBGEMM has recently added the support for Windows.

Differential Revision: [D19738268](https://our.internmc.facebook.com/intern/diff/D19738268/)

[ghstack-poisoned]

jianyuh committed:
Update on "[pt][fbgemm] Turn on USE_FBGEMM on Windows env"


As Title says. FBGEMM has recently added the support for Windows.

Differential Revision: [D19738268](https://our.internmc.facebook.com/intern/diff/D19738268/)

[ghstack-poisoned]

jianyuh committed:
Update on "[pt][fbgemm] Turn on USE_FBGEMM on Windows env"


As Title says. FBGEMM has recently added the support for Windows.

Differential Revision: [D19738268](https://our.internmc.facebook.com/intern/diff/D19738268/)

[ghstack-poisoned]

jianyuh committed:
Update on "[pt][fbgemm] Turn on USE_FBGEMM on Windows env"


As Title says. FBGEMM has recently added the support for Windows.

Differential Revision: [D19738268](https://our.internmc.facebook.com/intern/diff/D19738268/)

[ghstack-poisoned]

jianyuh committed:
Update on "[pt][fbgemm] Turn on USE_FBGEMM on Windows env"


As Title says. FBGEMM has recently added the support for Windows.

Differential Revision: [D19738268](https://our.internmc.facebook.com/intern/diff/D19738268/)

[ghstack-poisoned]

peterjc123 committed:
try fix fbgemm shared build

peterjc123 committed:
Fix typo

peterjc123 committed:
attempt 2

peterjc123 committed:
fix typo

peterjc123 committed:
Fix export on Windows

peterjc123 committed:
Remove comments

jianyuh committed:
Update on "[pt][fbgemm] Turn on USE_FBGEMM on Windows env"


As Title says. FBGEMM has recently added the support for Windows.

Differential Revision: [D19738268](https://our.internmc.facebook.com/intern/diff/D19738268/)

[ghstack-poisoned]

jianyuh committed:
Update on "[pt][fbgemm] Turn on USE_FBGEMM on Windows env"


As Title says. FBGEMM has recently added the support for Windows.

Differential Revision: [D19738268](https://our.internmc.facebook.com/intern/diff/D19738268/)

[ghstack-poisoned]

jianyuh committed:
Update on "[pt][fbgemm] Turn on USE_FBGEMM on Windows env"


As Title says. FBGEMM has recently added the support for Windows.

Differential Revision: [D19738268](https://our.internmc.facebook.com/intern/diff/D19738268/)

[ghstack-poisoned]

jianyuh committed:
Apply FBGEMM upgrade

jianyuh committed:
Update on "[pt][fbgemm] Turn on USE_FBGEMM on Windows env"


As Title says. FBGEMM has recently added the support for Windows.

Differential Revision: [D19738268](https://our.internmc.facebook.com/intern/diff/D19738268/)

[ghstack-poisoned]

jianyuh committed:
Update on "[pt][fbgemm] Turn on USE_FBGEMM on Windows env"


As Title says. FBGEMM has recently added the support for Windows.

Differential Revision: [D19738268](https://our.internmc.facebook.com/intern/diff/D19738268/)

[ghstack-poisoned]

jianyuh committed:
Update on "[pt][fbgemm] Turn on USE_FBGEMM on Windows env"


As Title says. FBGEMM has recently added the support for Windows.

Differential Revision: [D19738268](https://our.internmc.facebook.com/intern/diff/D19738268/)

[ghstack-poisoned]

peterjc123 commented on caffe2/quantization/server/elementwise_sum_dnnlowp_op_avx2.cc:
```diff
@@ -152,7 +153,7 @@ void ElementWiseSumAVX2(
 }
 
-template void ElementWiseSumAVX2<uint8_t, false>(
+template CAFFE2_API void ElementWiseSumAVX2<uint8_t, false>(
```
No, the template explicit instantiation doesn't need `CAFFE2_API`s. It should be added in the implementations, like in line 17.

jianyuh replied:
Thanks @peterjc123 ! This doesn't seem to work. Check the latest version of this PR. We still got the same errors for 
```
channel_shuffle_dnnlowp_op.cc.obj : error LNK2019: unresolved external symbol "void __cdecl fbgemm::transpose_4rows(int,unsigned char const *,unsigned char *)" (?transpose_4rows@fbgemm@@YAXHPEBEPEAE@Z) referenced in function "public: bool __cdecl caffe2::ChannelShuffleDNNLowPOp<unsigned char>::RunOnDeviceWithOrderNHWC$omp$1(void)" (?RunOnDeviceWithOrderNHWC$omp$1@?$ChannelShuffleDNNLowPOp@E@caffe2@@QEAA_NXZ)
elementwise_add_dnnlowp_op.cc.obj : error LNK2019: unresolved external symbol "void __cdecl caffe2::internal::ElementWiseSumAVX2<unsigned char,0>(unsigned char const *,unsigned char const *,unsigned char *,int,float,int,float,int,float,int)" (??$ElementWiseSumAVX2@E$0A@@internal@caffe2@@YAXPEBE0PEAEHMHMHMH@Z) referenced in function "public: virtual bool __cdecl caffe2::AddDNNLowPOp<unsigned char>::RunOnDevice$omp$1(void)" (?RunOnDevice$omp$1@?$AddDNNLowPOp@E@caffe2@@UEAA_NXZ)
elementwise_sum_dnnlowp_op.cc.obj : error LNK2001: unresolved external symbol "void __cdecl caffe2::internal::ElementWiseSumAVX2<unsigned char,0>(unsigned char const *,unsigned char const *,unsigned char *,int,float,int,float,int,float,int)" (??$ElementWiseSumAVX2@E$0A@@internal@caffe2@@YAXPEBE0PEAEHMHMHMH@Z)
elementwise_sum_dnnlowp_op.cc.obj : error LNK2019: unresolved external symbol "void __cdecl caffe2::internal::ElementWiseSumAVX2<unsigned char,1>(unsigned char const *,unsigned char const *,unsigned char *,int,float,int,float,int,float,int)" (??$ElementWiseSumAVX2@E$00@internal@caffe2@@YAXPEBE0PEAEHMHMHMH@Z) referenced in function "public: virtual bool __cdecl caffe2::SumDNNLowPOp<unsigned char,1>::RunOnDevice$omp$1(void)" (?RunOnDevice$omp$1@?$SumDNNLowPOp@E$00@caffe2@@UEAA_NXZ)
elementwise_sum_dnnlowp_op.cc.obj : error LNK2019: unresolved external symbol "void __cdecl caffe2::internal::ElementWiseSumAVX2<unsigned short,0>(unsigned short const *,unsigned short const *,unsigned short *,int,float,int,float,int,float,int)" (??$ElementWiseSumAVX2@G$0A@@internal@caffe2@@YAXPEBG0PEAGHMHMHMH@Z) referenced in function "public: virtual bool __cdecl caffe2::SumDNNLowPOp<unsigned short,0>::RunOnDevice$omp$1(void)" (?RunOnDevice$omp$1@?$SumDNNLowPOp@G$0A@@caffe2@@UEAA_NXZ)
elementwise_sum_dnnlowp_op.cc.obj : error LNK2019: unresolved external symbol "void __cdecl caffe2::internal::ElementWiseSumAVX2<unsigned short,1>(unsigned short const *,unsigned short const *,unsigned short *,int,float,int,float,int,float,int)" (??$ElementWiseSumAVX2@G$00@internal@caffe2@@YAXPEBG0PEAGHMHMHMH@Z) referenced in function "public: virtual bool __cdecl caffe2::SumDNNLowPOp<unsigned short,1>::RunOnDevice$omp$1(void)" (?RunOnDevice$omp$1@?$SumDNNLowPOp@G$00@caffe2@@UEAA_NXZ)
fully_connected_fake_lowp_op.cc.obj : error LNK2019: unresolved external symbol "void __cdecl caffe2::fp32_to_bfp16(float const *,unsigned __int64,float *)" (?fp32_to_bfp16@caffe2@@YAXPEBM_KPEAM@Z) referenced in function "public: bool __cdecl caffe2::FullyConnectedFakeLowpFPOp<&void __cdecl caffe2::fp32_to_bfp16(float const *,unsigned __int64,float *),class caffe2::CPUContext,class caffe2::DefaultEngine,1>::DoRunWithType<float,float,float,float,float>(void)" (??$DoRunWithType@MMMMM@?$FullyConnectedFakeLowpFPOp@$1?fp32_to_bfp16@caffe2@@YAXPEBM_KPEAM@ZVCPUContext@2@VDefaultEngine@2@$00@caffe2@@QEAA_NXZ)
fully_connected_fake_lowp_op.cc.obj : error LNK2019: unresolved external symbol "void __cdecl caffe2::fp32_to_bfp24(float const *,unsigned __int64,float *)" (?fp32_to_bfp24@caffe2@@YAXPEBM_KPEAM@Z) referenced in function "public: bool __cdecl caffe2::FullyConnectedFakeLowpFPOp<&void __cdecl caffe2::fp32_to_bfp24(float const *,unsigned __int64,float *),class caffe2::CPUContext,class caffe2::DefaultEngine,1>::DoRunWithType<float,float,float,float,float>(void)" (??$DoRunWithType@MMMMM@?$FullyConnectedFakeLowpFPOp@$1?fp32_to_bfp24@caffe2@@YAXPEBM_KPEAM@ZVCPUContext@2@VDefaultEngine@2@$00@caffe2@@QEAA_NXZ)
fully_connected_fake_lowp_op.cc.obj : error LNK2019: unresolved external symbol "void __cdecl caffe2::fp32_to_bfp14(float const *,unsigned __int64,float *)" (?fp32_to_bfp14@caffe2@@YAXPEBM_KPEAM@Z) referenced in function "public: bool __cdecl caffe2::FullyConnectedFakeLowpFPOp<&void __cdecl caffe2::fp32_to_bfp14(float const *,unsigned __int64,float *),class caffe2::CPUContext,class caffe2::DefaultEngine,1>::DoRunWithType<float,float,float,float,float>(void)" (??$DoRunWithType@MMMMM@?$FullyConnectedFakeLowpFPOp@$1?fp32_to_bfp14@caffe2@@YAXPEBM_KPEAM@ZVCPUContext@2@VDefaultEngine@2@$00@caffe2@@QEAA_NXZ)
fully_connected_fake_lowp_op.cc.obj : error LNK2019: unresolved external symbol "void __cdecl caffe2::fp32_to_fp16(float const *,unsigned __int64,float *)" (?fp32_to_fp16@caffe2@@YAXPEBM_KPEAM@Z) referenced in function "public: bool __cdecl caffe2::FullyConnectedFakeLowpFPOp<&void __cdecl caffe2::fp32_to_fp16(float const *,unsigned __int64,float *),class caffe2::CPUContext,class caffe2::DefaultEngine,1>::DoRunWithType<float,float,float,float,float>(void)" (??$DoRunWithType@MMMMM@?$FullyConnectedFakeLowpFPOp@$1?fp32_to_fp16@caffe2@@YAXPEBM_KPEAM@ZVCPUContext@2@VDefaultEngine@2@$00@caffe2@@QEAA_NXZ)
fully_connected_fake_lowp_op.cc.obj : error LNK2019: unresolved external symbol "void __cdecl caffe2::fp32_to_bfp16_round(float const *,unsigned __int64,float *)" (?fp32_to_bfp16_round@caffe2@@YAXPEBM_KPEAM@Z) referenced in function "public: bool __cdecl caffe2::FullyConnectedFakeLowpFPOp<&void __cdecl caffe2::fp32_to_bfp16_round(float const *,unsigned __int64,float *),class caffe2::CPUContext,class caffe2::DefaultEngine,1>::DoRunWithType<float,float,float,float,float>(void)" (??$DoRunWithType@MMMMM@?$FullyConnectedFakeLowpFPOp@$1?fp32_to_bfp16_round@caffe2@@YAXPEBM_KPEAM@ZVCPUContext@2@VDefaultEngine@2@$00@caffe2@@QEAA_NXZ)
...
```
in https://circleci.com/gh/pytorch/pytorch/4441479?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

cc @shz0116

shz0116 replied:
> Thanks @peterjc123 ! This doesn't seem to work. Check the latest version of this PR. We still got the same errors for
> 
> ```
> channel_shuffle_dnnlowp_op.cc.obj : error LNK2019: unresolved external symbol "void __cdecl fbgemm::transpose_4rows(int,unsigned char const *,unsigned char *)" (?transpose_4rows@fbgemm@@YAXHPEBEPEAE@Z) referenced in function "public: bool __cdecl caffe2::ChannelShuffleDNNLowPOp<unsigned char>::RunOnDeviceWithOrderNHWC$omp$1(void)" (?RunOnDeviceWithOrderNHWC$omp$1@?$ChannelShuffleDNNLowPOp@E@caffe2@@QEAA_NXZ)
> elementwise_add_dnnlowp_op.cc.obj : error LNK2019: unresolved external symbol "void __cdecl caffe2::internal::ElementWiseSumAVX2<unsigned char,0>(unsigned char const *,unsigned char const *,unsigned char *,int,float,int,float,int,float,int)" (??$ElementWiseSumAVX2@E$0A@@internal@caffe2@@YAXPEBE0PEAEHMHMHMH@Z) referenced in function "public: virtual bool __cdecl caffe2::AddDNNLowPOp<unsigned char>::RunOnDevice$omp$1(void)" (?RunOnDevice$omp$1@?$AddDNNLowPOp@E@caffe2@@UEAA_NXZ)
> elementwise_sum_dnnlowp_op.cc.obj : error LNK2001: unresolved external symbol "void __cdecl caffe2::internal::ElementWiseSumAVX2<unsigned char,0>(unsigned char const *,unsigned char const *,unsigned char *,int,float,int,float,int,float,int)" (??$ElementWiseSumAVX2@E$0A@@internal@caffe2@@YAXPEBE0PEAEHMHMHMH@Z)
> elementwise_sum_dnnlowp_op.cc.obj : error LNK2019: unresolved external symbol "void __cdecl caffe2::internal::ElementWiseSumAVX2<unsigned char,1>(unsigned char const *,unsigned char const *,unsigned char *,int,float,int,float,int,float,int)" (??$ElementWiseSumAVX2@E$00@internal@caffe2@@YAXPEBE0PEAEHMHMHMH@Z) referenced in function "public: virtual bool __cdecl caffe2::SumDNNLowPOp<unsigned char,1>::RunOnDevice$omp$1(void)" (?RunOnDevice$omp$1@?$SumDNNLowPOp@E$00@caffe2@@UEAA_NXZ)
> elementwise_sum_dnnlowp_op.cc.obj : error LNK2019: unresolved external symbol "void __cdecl caffe2::internal::ElementWiseSumAVX2<unsigned short,0>(unsigned short const *,unsigned short const *,unsigned short *,int,float,int,float,int,float,int)" (??$ElementWiseSumAVX2@G$0A@@internal@caffe2@@YAXPEBG0PEAGHMHMHMH@Z) referenced in function "public: virtual bool __cdecl caffe2::SumDNNLowPOp<unsigned short,0>::RunOnDevice$omp$1(void)" (?RunOnDevice$omp$1@?$SumDNNLowPOp@G$0A@@caffe2@@UEAA_NXZ)
> elementwise_sum_dnnlowp_op.cc.obj : error LNK2019: unresolved external symbol "void __cdecl caffe2::internal::ElementWiseSumAVX2<unsigned short,1>(unsigned short const *,unsigned short const *,unsigned short *,int,float,int,float,int,float,int)" (??$ElementWiseSumAVX2@G$00@internal@caffe2@@YAXPEBG0PEAGHMHMHMH@Z) referenced in function "public: virtual bool __cdecl caffe2::SumDNNLowPOp<unsigned short,1>::RunOnDevice$omp$1(void)" (?RunOnDevice$omp$1@?$SumDNNLowPOp@G$00@caffe2@@UEAA_NXZ)
> fully_connected_fake_lowp_op.cc.obj : error LNK2019: unresolved external symbol "void __cdecl caffe2::fp32_to_bfp16(float const *,unsigned __int64,float *)" (?fp32_to_bfp16@caffe2@@YAXPEBM_KPEAM@Z) referenced in function "public: bool __cdecl caffe2::FullyConnectedFakeLowpFPOp<&void __cdecl caffe2::fp32_to_bfp16(float const *,unsigned __int64,float *),class caffe2::CPUContext,class caffe2::DefaultEngine,1>::DoRunWithType<float,float,float,float,float>(void)" (??$DoRunWithType@MMMMM@?$FullyConnectedFakeLowpFPOp@$1?fp32_to_bfp16@caffe2@@YAXPEBM_KPEAM@ZVCPUContext@2@VDefaultEngine@2@$00@caffe2@@QEAA_NXZ)
> fully_connected_fake_lowp_op.cc.obj : error LNK2019: unresolved external symbol "void __cdecl caffe2::fp32_to_bfp24(float const *,unsigned __int64,float *)" (?fp32_to_bfp24@caffe2@@YAXPEBM_KPEAM@Z) referenced in function "public: bool __cdecl caffe2::FullyConnectedFakeLowpFPOp<&void __cdecl caffe2::fp32_to_bfp24(float const *,unsigned __int64,float *),class caffe2::CPUContext,class caffe2::DefaultEngine,1>::DoRunWithType<float,float,float,float,float>(void)" (??$DoRunWithType@MMMMM@?$FullyConnectedFakeLowpFPOp@$1?fp32_to_bfp24@caffe2@@YAXPEBM_KPEAM@ZVCPUContext@2@VDefaultEngine@2@$00@caffe2@@QEAA_NXZ)
> fully_connected_fake_lowp_op.cc.obj : error LNK2019: unresolved external symbol "void __cdecl caffe2::fp32_to_bfp14(float const *,unsigned __int64,float *)" (?fp32_to_bfp14@caffe2@@YAXPEBM_KPEAM@Z) referenced in function "public: bool __cdecl caffe2::FullyConnectedFakeLowpFPOp<&void __cdecl caffe2::fp32_to_bfp14(float const *,unsigned __int64,float *),class caffe2::CPUContext,class caffe2::DefaultEngine,1>::DoRunWithType<float,float,float,float,float>(void)" (??$DoRunWithType@MMMMM@?$FullyConnectedFakeLowpFPOp@$1?fp32_to_bfp14@caffe2@@YAXPEBM_KPEAM@ZVCPUContext@2@VDefaultEngine@2@$00@caffe2@@QEAA_NXZ)
> fully_connected_fake_lowp_op.cc.obj : error LNK2019: unresolved external symbol "void __cdecl caffe2::fp32_to_fp16(float const *,unsigned __int64,float *)" (?fp32_to_fp16@caffe2@@YAXPEBM_KPEAM@Z) referenced in function "public: bool __cdecl caffe2::FullyConnectedFakeLowpFPOp<&void __cdecl caffe2::fp32_to_fp16(float const *,unsigned __int64,float *),class caffe2::CPUContext,class caffe2::DefaultEngine,1>::DoRunWithType<float,float,float,float,float>(void)" (??$DoRunWithType@MMMMM@?$FullyConnectedFakeLowpFPOp@$1?fp32_to_fp16@caffe2@@YAXPEBM_KPEAM@ZVCPUContext@2@VDefaultEngine@2@$00@caffe2@@QEAA_NXZ)
> fully_connected_fake_lowp_op.cc.obj : error LNK2019: unresolved external symbol "void __cdecl caffe2::fp32_to_bfp16_round(float const *,unsigned __int64,float *)" (?fp32_to_bfp16_round@caffe2@@YAXPEBM_KPEAM@Z) referenced in function "public: bool __cdecl caffe2::FullyConnectedFakeLowpFPOp<&void __cdecl caffe2::fp32_to_bfp16_round(float const *,unsigned __int64,float *),class caffe2::CPUContext,class caffe2::DefaultEngine,1>::DoRunWithType<float,float,float,float,float>(void)" (??$DoRunWithType@MMMMM@?$FullyConnectedFakeLowpFPOp@$1?fp32_to_bfp16_round@caffe2@@YAXPEBM_KPEAM@ZVCPUContext@2@VDefaultEngine@2@$00@caffe2@@QEAA_NXZ)
> ...
> ```
> 
> in https://circleci.com/gh/pytorch/pytorch/4441479?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link
> 
> cc @shz0116

This seems not an EXPORT issue. I doubt somewhere the transpose.cc not linked. Could you try to #include "transpose.cc" directly to see how it works ?

jianyuh committed:
Update on "[pt][fbgemm] Turn on USE_FBGEMM on Windows env"


As Title says. FBGEMM has recently added the support for Windows.

Differential Revision: [D19738268](https://our.internmc.facebook.com/intern/diff/D19738268/)

[ghstack-poisoned]
